### PR TITLE
Fix footer styling in docs

### PR DESCRIPTION
**Pull request by bgunnar5**
Added configuration files including .readthedocs.yaml and docs/requirements.txt to get sphinx updated to the newest version. This fixed the styling issue with the footer and also made docs/source/_static/theme_overrides.css unnecessary so it was removed. Updating sphinx caused issues with the Makefile in docs/ so I modified it by adding a pip install command and a new clean method. 

There was a secondary issue involving the variables page in the docs where the horizontal scroll was overflowing behind the left nav bar. This was also patched by modifying the custom.css file and slightly modifying how the reserved variables table in merlin_variables.rst was formatted.

### CODE REVIEW

File: docs/source/merlin_variables.rst [hunk 1]
```diff
@@ -31,73 +31,138 @@ The directory structure of merlin output looks like this:
...
-  * - ``$(MERLIN_SPEC_ORIGINAL_TEMPLATE)``
-    - Copy of original yaml file passed to ``merlin run``.
-    - ``$(MERLIN_INFO)/*.orig.yaml``
-  * - ``$(MERLIN_SPEC_EXECUTED_RUN)``
-    - Parsed and processed yaml file with command-line variable substitutions included.
-    - ``$(MERLIN_INFO)/*.partial.yaml``
-  * - ``$(MERLIN_SPEC_ARCHIVED_COPY)``
-    - Archive version of ``MERLIN_SPEC_EXECUTED_RUN`` with all variables and paths fully resolved.
-    - ``$(MERLIN_INFO)/*.expanded.yaml``
+   :widths: 25 50 25
```
Comment (lucpeterson) [inline docs/source/merlin_variables.rst:34]: is this working as intended?

Comment (bgunnar5) [reply docs/source/merlin_variables.rst:34]: I added that :widths: tag prior to modifying the css for tds to have a max-width of 300px so there's a chance it's not needed anymore. Let me go check.

Comment (bgunnar5) [reply docs/source/merlin_variables.rst:34]: It looks like that's not even needed with the new custom css. I'll remove it and push again.

**Comment by lucpeterson** [comment]
Superseded by #383

**Pull request closed without merging**